packages/ember-metal/lib/weak_map.js

```diff
@@ -20,13 +24,19 @@ function UNDEFINED() {}
  * practice, most use cases satisfy this limitation which is why it is included
  * in ember-metal.
  */
-export default function WeakMap() {
-  assert(
-    'Invoking the WeakMap constructor with arguments is not supported at this time',
-    arguments.length === 0
-  );
-
+export default function WeakMap(iterable) {
   this._id = GUID_KEY + (id++);
+
+  if (iterable === null || iterable === undefined) {
+    return;
+  } else if (Array.isArray(iterable)) {
```

Comment:
I will also update https://github.com/thoov/ember-weakmap to match this once it gets merged